pkg/logs/config/processing_rules.go

```diff
@@ -0,0 +1,85 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package config
+
+import (
+	"fmt"
+	"regexp"
+)
+
+// Processing rule types
+const (
+	ExcludeAtMatch = "exclude_at_match"
+	IncludeAtMatch = "include_at_match"
+	MaskSequences  = "mask_sequences"
+	MultiLine      = "multi_line"
+)
+
+// ProcessingRule defines an exclusion or a masking rule to
+// be applied on log lines
+type ProcessingRule struct {
+	Type               string
+	Name               string
+	ReplacePlaceholder string `mapstructure:"replace_placeholder" json:"replace_placeholder"`
+	Pattern            string
+	// computed properties
+	Regex       *regexp.Regexp
```

Comment:
I'd still love to see a proper type (CompiledRule?) that holds a reference to the config and the compiled regex, rather than everything bundled together here. Can we at least keep the // TODO ?

Reply:
done